examples/routing_service/udp_socket_adapter/src/SocketAdapter.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+ *
+ * RTI grants Licensee a license to use, modify, compile, and create derivative
+ * works of the Software.  Licensee has the right to distribute object form
+ * only for use with RTI products.  The Software is provided "as is", with no
+ * warranty of any type, including any warranty for fitness for any purpose.
+ * RTI is under no obligation to maintain or support the Software.  RTI shall
+ * not be liable for any incidental or consequential damages arising out of the
+ * use or inability to use the software.
+ */
+
+#ifndef SOCKETADAPTER_HPP
+#define SOCKETADAPTER_HPP
+
+#include <dds/core/corefwd.hpp>
+#include <dds/core/macros.hpp>
+#include <rti/routing/PropertySet.hpp>
+#include <rti/routing/RoutingService.hpp>
+#include <rti/routing/adapter/AdapterPlugin.hpp>
+
+class SocketAdapter : public rti::routing::adapter::AdapterPlugin {
```

Comment:
This is example code. Let's add a comment explaining this class. E.g. make clear that the code is a read-only adapter, etc

Reply:
We should make sure the users understand that the socket adapter is hard-coded to read `ShapeType`. They should understand the limitations and usage.

Reply:
Another good addition would be tips on what to do to tune the example for their own types.
Also, do we recommend using type generator, but having it not generate writer/reader code, just the type, and link it here to the adapter?
Should we actually modify the example so we do exactly that with `ShapeType`?

Reply:
- I added a comment explaining the class
- In the README, I added a couple of lines explaining that ShapeType and Square are hardcoded, and that if they want to build a real adapter, they should modify the code and XML accordingly
- I don't agree with using the generated types here, I don't think it adds too much value. This is a non-DDS data to DDS adapter. The data coming from the socket (non-DDS) most likely doesn't have IDL / CDR format, so I don't see how including here the extra step of running rtiddsgen adds value. The user needs to use DynamicData. I do see value if this was a processor or a transformation